File: ios/FamilyMovieNight/Models/WatchedMovie.swift
import Foundation

struct WatchedMoviesResponse: Codable {
    let watchedMovies: [WatchedMovie]
}

struct WatchedMovie: Codable, Identifiable {
    let tmdbMovieId: Int
    let title: String
    let posterPath: String
    let year: Int
    let watchedAt: String
    let source: String
    let markedBy: String
    let pickId: String?
    let avgRating: Double?

    var id: Int { tmdbMovieId }

    var posterURL: URL? {
        guard !posterPath.isEmpty else { return nil }
        return URL(string: "https://image.tmdb.org/t/p/w342\(posterPath)")
    }
}

struct MarkWatchedRequest: Encodable {
    let tmdbMovieId: Int
}

File: backend/src/models/watched-movie.ts
import { z } from "zod";

export const WatchedMovieSchema = z.object({
  group_id: z.string(),
  tmdb_movie_id: z.number(),
  marked_by: z.string(),
  watched_at: z.string(),
  title: z.string(),
  poster_path: z.string(),
  year: z.number(),
  source: z.literal("direct"),
});

export const MarkWatchedDirectSchema = z.object({
  tmdb_movie_id: z.number({ required_error: "tmdb_movie_id is required" }),
  title: z.string({ required_error: "title is required" }),
  poster_path: z.string({ required_error: "poster_path is required" }),
  year: z.number({ required_error: "year is required" }),
});

export interface CombinedWatchedMovie {
  tmdb_movie_id: number;
  title: string;
  poster_path: string;
  year: number;
  watched_at: string;
  source: "picked" | "direct";
  marked_by: string;
  pick_id?: string;
  avg_rating?: number;
}

export type WatchedMovie = z.infer<typeof WatchedMovieSchema>;
export type MarkWatchedDirectInput = z.infer<typeof MarkWatchedDirectSchema>;
